hadoop-crypto/src/test/java/com/palantir/hadoop/cipher/AbstractFsCipherInputStreamTest.java

```diff
@@ -165,8 +166,11 @@ public final void testBulkRead() throws IOException {
         byte[] buffer = new byte[NUM_BYTES];
         int offset = 0;
         int read;
-        while ((read = cis.read(buffer, offset, buffer.length)) != -1) {
+        int len = buffer.length;
+        while ((read = cis.read(buffer, offset, len)) != 0) {
             offset += read;
+            len -= read;
```

Comment:
Why do we adjust the len here? If we didn't read the full buffer length, the next time through should try to read from offset to buffer.length